src/Sema.zig

```diff
@@ -13968,9 +13972,29 @@ fn zirImport(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!Air.
             return sema.fail(block, operand_src, "unable to open '{s}': {s}", .{ operand, @errorName(err) });
         },
     };
-    try pt.ensureFileAnalyzed(result.file_index);
-    const file_root_decl_index = zcu.fileRootDecl(result.file_index).unwrap().?;
-    return sema.analyzeDeclVal(block, operand_src, file_root_decl_index);
+    switch (result.file.mode) {
+        .zig => {
+            try pt.ensureFileAnalyzed(result.file_index);
+            const file_root_decl_index = zcu.fileRootDecl(result.file_index).unwrap().?;
+            return sema.analyzeDeclVal(block, operand_src, file_root_decl_index);
+        },
+        .zon => {
+            _ = result.file.getTree(zcu.gpa) catch |err| {
+                // TODO: these errors are file system errors; make sure an update() will
```

Comment:
This TODO was already present in other instances of similar code in this file, so I kept it.